### PR TITLE
CLICKHOUSE-3772 + CLICKHOUSE-3791: Fixes and new columns in system tables formats, data_type_families, aggregate_function_combinators

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: dbms/src/Storages/System/StorageSystemDataTypeFamilies.cpp [hunk 1]
```diff
@@ -56,35 +53,60 @@ namespace
...
+                if (type_ptr->haveMaximumSizeOfValue())
+                    size = type_ptr->getMaximumSizeOfValueInMemory();
             }
 
-            res_columns[1]->insert(getPropertiesAsString(type_ptr));
+            res_columns[3]->insert(size);
+
+            setTypePropertries(type_ptr, res_columns);
         }
         catch (Exception & ex)
```
Comment (alexey-milovidov) [inline dbms/src/Storages/System/StorageSystemDataTypeFamilies.cpp]: const

Comment (alexey-milovidov) [reply dbms/src/Storages/System/StorageSystemDataTypeFamilies.cpp]: ex is not used

File: dbms/src/Storages/System/StorageSystemDataTypeFamilies.cpp [hunk 2]
```diff
@@ -56,35 +53,60 @@ namespace
...
         {
-            res_columns[1]->insert(String{"depends_on_arguments"});
+            setComplexTypeProperties(name, res_columns);
```
Comment (alexey-milovidov) [inline dbms/src/Storages/System/StorageSystemDataTypeFamilies.cpp]: It's wrong to use exceptions for non-exceptional control flow.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: dbms/src/Storages/System/StorageSystemDataTypeFamilies.cpp (hunk 2 above)
Comment (alesapin) [reply dbms/src/Storages/System/StorageSystemDataTypeFamilies.cpp]: I have thought about this. But how can we know, that the type is complex/have nested types if we would not try to instantiate it? I see only one solution -- hardcoded types names, but I think current logic is better.

**Comment by alesapin** [comment]
Locally all tests are passing:
```
./clickhouse-test 00534

Running stateless tests.

00534_client_ignore_error:                                             [ OK ]
00534_exp10:                                                           [ OK ]
00534_filimonov:                                                       [ OK ]
00534_long_functions_bad_arguments1:                                   [ OK ]
00534_long_functions_bad_arguments2:                                   [ OK ]
00534_long_functions_bad_arguments3:                                   [ OK ]
00534_long_functions_bad_arguments4:                                   [ OK ]
00534_long_functions_bad_arguments5:                                   [ OK ]
00534_long_functions_bad_arguments6:                                   [ OK ]
00534_long_functions_bad_arguments7:                                   [ OK ]
00534_long_functions_bad_arguments8:                                   [ OK ]
00534_long_functions_bad_arguments9:                                   [ OK ]

12 tests passed. 0 tests skipped.
```

**Pull request merged**